### DISCUSSION TITLE
Defined currency is not taken over in the report and in the payment section

### DISCUSSION THREAD

**Discussion by ciucaadrian11** [Intermesh/groupoffice#996] [Q&A]
Hello,

I detected in the Finance module as the currency that I set in the book, at invoice, purchase invoice, etc. is not taken over in the report and payment. Please consult de bellow print screen. Did I miss something?

![image](https://github.com/Intermesh/groupoffice/assets/57172223/6f788d3e-10bf-42df-8129-032aa9ef6f6a)

From the book "Factura (RON)" I issued invoices in "Lei" currency and from the book "Factura (EURO) " II issued invoices in "EURO" currency. It seems at the payment section (right sidebar of the invoice) is displayed the currency defined in "My Account".
 
![image](https://github.com/Intermesh/groupoffice/assets/57172223/b42e6ff2-16d1-428f-bb8b-7657c94be745)

In report:
![image](https://github.com/Intermesh/groupoffice/assets/57172223/788c872f-d421-41e3-9599-16f9e35ab0ae)

Same thing is happened also for the articles from catalog
![image](https://github.com/Intermesh/groupoffice/assets/57172223/8cd56d74-9606-4049-8d75-bd164524faa0)
![image](https://github.com/Intermesh/groupoffice/assets/57172223/b159c988-1265-4c73-a17e-80bbee5046b6)

Also, I don't know how is more better to proceed if I need to invoiced in two currencies. Lets to take an exemple: I have customers from EU and also customers from US. For the customers from UE I have to issue invoices in EUR and for the customers from US I have to issue invoices in US Dollar. The solution that I found currently is to make two tipes of books as "Sales invoices". and to define the currency for each book (Euro and US Dollar). It is little bit inconvenient to keet the invoices in two separate books. As an ideea: it is not more efficient to set the currency at the products/services difeind in catalog? Or to chose the billing currency when is creating the invoice? In this way is not necessary to keep the invoices in two separate book. Or is there another solution that I haven't discovered yet?

I am still discovering this beautiful software.
Thank you very much!

**Comment by mschering** [comment]
Hi,

Thanks, I will fix these issues with the display.

If we put separate currencies in the same book. How could we deal with the exchange rates and reports of total income etc.?

**Reply by ciucaadrian11** [reply]
Thank you for your support! And also, I think is better to be displayed in the report the total amount, not the subtotal amount. 

"If we put separate currencies in the same book. How could we deal with the exchange rates and reports of total income etc.?" It was just an initial ideea. But I think there are so many company who have to invoice in several currencies.

**Comment by michalcharvat** [comment]
In common accounting programs there is invoice in a foreign currency. However on the same invoice you have to calculate amount and sometimes also a VAT for tax office. I assume the best would be add there an option create exchange rate in invoice dialog or add option use exchange API - for example: https://www.cnb.cz/cs/financni-trhy/devizovy-trh/kurzy-devizoveho-trhu/kurzy-devizoveho-trhu/denni_kurz.txt however it does mean you have to define somewhere url and i dont know user defined lamba function that will define how do you parse the data from API - maybe submodule for financial module thats allows you add exchange rates similar to payment methods? Maybe you can read exchange rates directly from ECB - at least for EU countries

**Comment by mschering** [comment]
Not sure I fully understand. Do you mean for example:

- Default currency is €
- All reporting is in €
- You can set an alternate currency for a single invoice. For example USD
- You enter all pricing in USD.
- The exchange rate will be taken from some API and in the database the exchange rate is stored and the values are stored in € so you have all your totals and reporting in €.

**Reply by michalcharvat** [reply]
Basically yes. Of course you can have two accounts (€, $) however from point of view tax office you have to always calculate with the amount from the date when did you create the invoice so I assume the report should also display values in default currency. There are basically two options how to convert value on invoice to default currency - with exchange date that was in the date 1) when you created the invoice or date of taxable supply or 2) the average of the CNB exchange rates on the last day of each month of the year (published in the financial newsletter). So from this point of view - there could be two options in settings - fixed rate or rate loaded from API
